From 2f455a8a50cbca16ae6fdfbbf2fd27484e468855 Mon Sep 17 00:00:00 2001
From: Dave Young <dyoung@redhat.com>
Date: Thu, 22 May 2014 02:43:19 -0400
Subject: [x86] ksysfs.c build fix

Message-id: <1400726607-13717-28-git-send-email-dyoung@redhat.com>
Patchwork-id: 80128
O-Subject: [RHEL7.1 PATCH 27/35] x86: ksysfs.c build fix
Bugzilla: 1080109
RH-Acked-by: Chao WANG <chaowang@redhat.com>
RH-Acked-by: Vivek Goyal <vgoyal@redhat.com>

Resolves: bz1080109

Brew build:
https://brewweb.devel.redhat.com/taskinfo?taskID=7480539

Test: Tested kexec/kdump on HP workstation and qemu.

Upstream commit:
commit 41a34cec2e0df7798ea322ed1480bc3d3facdc8e
Author: Dave Young <dyoung@redhat.com>
Date:   Fri Jan 3 11:54:31 2014 +0800

    x86: ksysfs.c build fix

    kbuild test robot report below error for randconfig:

      arch/x86/kernel/ksysfs.c: In function 'get_setup_data_paddr':
      arch/x86/kernel/ksysfs.c:81:3: error: implicit declaration of function 'ioremap_cache' [-Werror=implicit-function-declara
      arch/x86/kernel/ksysfs.c:86:3: error: implicit declaration of function 'iounmap' [-Werror=implicit-function-declaration]

    Fix it by including <asm/io.h> in ksysfs.c

    Signed-off-by: Dave Young <dyoung@redhat.com>
    Signed-off-by: Matt Fleming <matt.fleming@intel.com>

Signed-off-by: Jarod Wilson <jarod@redhat.com>

diff --git a/arch/x86/kernel/ksysfs.c b/arch/x86/kernel/ksysfs.c
index eb53d15..c2bedae 100644
--- a/arch/x86/kernel/ksysfs.c
+++ b/arch/x86/kernel/ksysfs.c
@@ -17,6 +17,7 @@
 #include <linux/slab.h>
 #include <linux/mm.h>
 
+#include <asm/io.h>
 #include <asm/setup.h>
 
 static ssize_t version_show(struct kobject *kobj,
-- 
1.7.1